### PR TITLE
Fix potential feerate overflow issues

### LINKED ISSUE #327: Make FeeRate arithmetic operations safe
See unresolved comments in the original [mempool PR](https://github.com/mintlayer/mintlayer-core/pull/43).

### PR DESCRIPTION
**Pull request by royster57**
This PR is meant to close #327. @iljakuklic and @TheQuantumPhysicist, let me know what you think.

### CODE REVIEW

File: mempool/src/error.rs [hunk 1]
```diff
@@ -81,6 +81,8 @@ pub enum TxValidationError {
     RollingFeeThresholdNotMet { minimum_fee: Amount, tx_fee: Amount },
     #[error("Overflow encountered while updating ancestor fee.")]
     AncestorFeeUpdateOverflow,
+    #[error("Fee overflow")]
+    FeeOverflow,
```
Comment (stanislav-tkach) [inline mempool/src/error.rs:85]: I'm just curious if we can have a more specific error (such as `AncestorFeeUpdateOverflow` above) or there are several completely different places where the new error can occur?..

**Review by stanislav-tkach — APPROVED** [review]

File: mempool/src/error.rs (hunk 1 above)
Comment (royster57) [reply mempool/src/error.rs:85]: The new error occurs in just one place. But calling it `FeePrecisionMultiplicationOverflow` or something of the sort seemed over the top....

File: mempool/src/feerate.rs [hunk 2]
```diff
@@ -51,11 +62,12 @@ impl FeeRate {
         self.atoms_per_kb
     }
 
-    // TODO Use NonZeroUsize for divisor
-    fn div_up(dividend: u128, divisor: usize) -> u128 {
-        debug_assert!(divisor != 0);
-        let divisor = u128::try_from(divisor).expect("div_up conversion");
-        (dividend + divisor - 1) / divisor
+    fn div_up(dividend: Amount, divisor: NonZeroUsize) -> Amount {
```
Comment (TheQuantumPhysicist) [inline mempool/src/feerate.rs:65]: Please call this `div_and_round_up` or something that indicates that it's dividing then rounding up. `div_up` is not an agreed upon naming standard that I know of.

**Review by TheQuantumPhysicist — COMMENTED** [review]
Can you please review bitcoin's feerate.cpp and compare with your implementations? For some reason I see you're not rounding as they're doing.

File: mempool/src/feerate.rs (hunk 2 above)
Comment (royster57) [reply mempool/src/feerate.rs]: Following your other comment, I changed our implementation to follow Bitcoin Core (there was no good reason why I diverged), and the need for this function disappeared.

Comment (royster57) [reply mempool/src/feerate.rs]: I also changed `FeeRate`'s underlying type from `u128` to `Amount`.

**Review by TheQuantumPhysicist — APPROVED** [review]
Looks good.

**Pull request merged**